### DISCUSSION TITLE
Consuming swup as native ESM module

### DISCUSSION THREAD

**Discussion by dgpapagian** [swup/swup#804] [Help] [answered]
Uncaught SyntaxError: Cannot use import statement outside a module (at Swup.modern.js:1:1)

**Comment by daun** [comment]
This error usually shows up when importing native ES modules from a non-module context.

Check out this article for possible pointers: [How to fix "cannot use import statement outside a module"](https://flaviocopes.com/fix-cannot-use-import-outside-module/).

**Reply by dgpapagian** [reply]
When I declare the swup as module (`<script defer type="module" src="node_modules/swup/dist/Swup.modern.js"></script>`) I get this error: 

Uncaught TypeError: Failed to resolve module specifier "delegate-it". Relative references must start with either "/", "./", or "../".

Also, I cannot refer it to my index.js as i get `Uncaught ReferenceError: Swup is not defined`

**Reply by dgpapagian** [reply]
Seems to work when I use the umd.js, but why is that?

**Reply by daun** [reply, marked as answer]
That's the browser trying to load one of swup's dependencies (`delegate-it`), but not knowing where to load it from. The module and modern builds don't bundle dependencies so you can set up [import maps](https://www.honeybadger.io/blog/import-maps/) or any other type of bundling yourself. The UMD build, on the other hand, comes with all dependencies already inlined to allow quick prototyping from simple `script` tags.

The simplest way to consume swup as ES module is using a CDN that knows about ES modules like unpkg. See [this example](https://swup.js.org/getting-started/installation/#esm) from the docs on ESM. It uses [unpkg's module param](https://www.unpkg.com/#query-params) to inline any dependencies on-the-fly.

```html
<script type="module">
  import Swup from 'https://unpkg.com/swup@4?module';
  const swup = new Swup();
</script>
```

**Reply by daun** [reply]
Let us know if this helped solve your problem, @dgpapagian.
